packages/db/src/indexes/auto-index.ts

```diff
@@ -14,6 +14,19 @@ function shouldAutoIndex(collection: CollectionImpl<any, any, any, any, any>) {
     return false
   }
 
+  // Check if defaultIndexType is set on the collection
+  if (!collection.config.defaultIndexType) {
+    if (isDevModeEnabled()) {
+      console.warn(
+        `[TanStack DB] Auto-indexing is enabled but no defaultIndexType is set. ` +
```

Comment:
nit: this warning is misleading because it says auto-indexing is enabled but actually since we return false here it is disabled. We should make explicit that we are disabling it because no defaultIndexType is set.

I think it would make more sense to still keep it enabled and just fall back to a default index type in case the user does not provide one?